### PR TITLE
Unregister packages when a cabal file is no longer referenced

### PR DESCRIPTION
**Pull request by jacobstanley**

### CODE REVIEW

File: main/mafia.hs [hunk 1]
```diff
@@ -327,20 +328,65 @@ mkFileMap = Map.fromList . fmap (\path -> (takeFileName path, path))
...
-     -> do liftIO (T.hPutStrLn stderr ("Cache: Removed " <> takeFileName file))
+    Delete file -> do
+      liftIO (T.hPutStrLn stderr ("Cache: Removed " <> takeFileName file))
 
-    Update src _
-     -> do rel <- fromMaybe src <$> makeRelativeToCurrentDirectory src
-           liftIO (T.hPutStrLn stderr ("Cache: Modified " <> rel))
+    Update src _ -> do
+      rel <- fromMaybe src <$> makeRelativeToCurrentDirectory src
+      liftIO (T.hPutStrLn stderr ("Cache: Modified " <> rel))
```
Comment (jacobstanley) [inline main/mafia.hs:336]: ^ sorry this is just a style fixup so all the case expressions are the same in here

**Comment by tranma** [comment]
does this unregister dependencies when they are removed from the .cabal file?

**Comment by markhibberd** [comment]
:8ball:

**Comment by jacobstanley** [comment]
@tranma no it's only for submodule dependencies

**Comment by jacobstanley** [comment]
Hmm I think this might have a few broken corner cases, it's one of the reasons I wrote the test in the other PR. I'm going to expand that test to include messing with the `<project>.submodules` file and then see if this still works.

**Comment by jacobstanley** [comment]
I'm going to put up a different PR for this which actually works, and has more thorough tests.
